controller/sync.go

```diff
@@ -490,24 +506,41 @@ func getMergePatch(original, modified *unstructured.Unstructured, lookupPatchMet
 		return nil, err
 	}
 	if lookupPatchMeta != nil {
-		return strategicpatch.CreateThreeWayMergePatch(modifiedJSON, modifiedJSON, originalJSON, lookupPatchMeta, true)
+		return strategicpatch.CreateThreeWayMergePatch(originalJSON, modifiedJSON, originalJSON, lookupPatchMeta, true)
```

Comment:
Why does this change from `modifiedJSON` to `originalJSON`?

Reply:
I ran the unit tests and they were unaffected when switching it back to the previous impl - so my question remains, why do we need to modify the merge patch? 

Can you add a test which needs this (alternatively create a test that fails if this were to be modified)?